### DISCUSSION TITLE
Extract Address from Rotated PDF in proper format

### DISCUSSION THREAD

**Discussion by tchoffman** [jsvine/pdfplumber#1048] [Ask for help with specific PDFs]
[John and Jane Doe 0.pdf](https://github.com/jsvine/pdfplumber/files/13480509/John.and.Jane.Doe.0.pdf)
[John and Jane Doe 90.pdf](https://github.com/jsvine/pdfplumber/files/13480512/John.and.Jane.Doe.90.pdf)

I'm attempting to crop a portion of a PDF and then extract multiline text from that cropped PDF.  As an example, somewhere on a PDF I may be trying to extract the text of a 3 line address. In a normally oriented (`John and Jane Doe 0.pdf`) PDF I can simply crop the area and use `extract_text()` and yield:
```
John and Jane Doe
1234 Broadway Avenue
Beverly Hills, CA 90210
```
However, when I attempt to crop the area around the address and use `extract_text()` on a rotated PDF (such as `John and Jane Doe 90.pdf`) I get:

```
Beverly
Hills,
CA
90210
1234
Broadway
Avenue
John
and
Jane
Doe
```

This is because the last line of the address is to the left of the 2nd and 1st lines.  I've tried using `extract_text(horizonal_ltr=True)` but I get the same output.  Is there a way to notify the text extraction that the cropped PDF is rotated and should be adjusted?  Or is there a way to manipulate the output in a way that I can order the text in the correct order?  For instance, is there some data structure that contains a 2-D array of words for instance that I can easily rotate back?  Or, can I apply rotation to `Page` objects?

**Comment by jsvine** [comment]
Thanks for sharing this use-case, @tchoffman. Currently, `pdfplumber` doesn't handle rotated pages very well; or, rather, it doesn't provide specific features for handling them. That might change with the proposals in https://github.com/jsvine/pdfplumber/pull/1040. 

> Or is there a way to manipulate the output in a way that I can order the text in the correct order? For instance, is there some data structure that contains a 2-D array of words for instance that I can easily rotate back?

I'm not sure if this answers your question / satisfies your need, but you might try the `.extract_words(...)` method, which should return an array of objects, each representing a word on the page — its text, positioning, etc.

**Reply by tchoffman** [reply]
Thanks for the reply. I've tried `extract_words()` briefly. I'll give a deeper look. And maybe I'll put a comment on that PR. At first glance it seems similar to what we're trying to do.
